common/setups/rasr/util/nn.py

```diff
@@ -198,52 +194,116 @@ def get_data_dict(self):
         }
 
 
+# Attribute names are invalid identifiers, therefore use old syntax
+SearchParameters = TypedDict(
+    "SearchParameters",
+    {
+        "beam-pruning": float,
+        "beam-pruning-limit": float,
+        "lm-state-pruning": Optional[float],
+        "word-end-pruning": float,
+        "word-end-pruning-limit": float,
+    },
+)
+
+
+class LookaheadOptions(TypedDict):
+    cache_high: Optional[int]
+    cache_low: Optional[int]
+    history_limit: Optional[int]
+    laziness: Optional[int]
+    minimum_representation: Optional[int]
+    tree_cutoff: Optional[int]
+
+
+class LatticeToCtmArgs(TypedDict):
+    best_path_algo: Optional[str]
+    encoding: Optional[str]
+    extra_config: Optional[Any]
+    extra_post_config: Optional[Any]
+    fill_empty_segments: Optional[bool]
+
+
+class NnRecogArgs(TypedDict):
+    acoustic_mixture_path: Optional[tk.Path]
```

Comment:
I noticed an inconsistency...
`acoustic_mixture_path` and `acoustic_mixtures` for the same kind of object... I think I introduced that inconsistency...
@michelwi @vieting @JackTemaki @Marvin84 preferences?

Reply:
Address this in a new PR or here?

Reply:
@JackTemaki your opinion?

Reply:
New issue.